src/js/angular/ttyg/controllers/ttyg-view.controller.js

```diff
@@ -730,6 +728,32 @@ function TTYGViewCtrl(
         $window.open('/connectors', '_blank');
     };
 
+    /**
+     * Opens SPARQL editor view with passed query.
+     * @param {{query: string, repositoryId: string}} payload
+     */
+    const onGoToSparqlEditorView = (payload) => {
+        if (payload.repositoryId !== $repositories.getActiveRepository()) {
+            const repository = $repositories.getRepository(payload.repositoryId);
+            if (repository) {
+                ModalService.openConfirmation(
+                    $translate.instant('common.confirm'),
+                    decodeHTML($translate.instant('ttyg.chat_panel.dialog.confirm_repository_change.body', {repositoryId: payload.repositoryId})),
+                    () => {
+                        $repositories.setRepository(repository);
```

Comment:
what does this do? I should test it

Reply:
AFAICS this effectively changes the currently selected repository and notifies interested parts about that. Im not sure we have to do that

Reply:
We discussed this behavior with the team. We ask the user for confirmation, and if they confirm, we change the repository and continue.

Reply:
But this would change it for the current tab as well. Also, this might affect the agent creation process, because by openning the create dialog, the repository id is set to current repository, which might be confusing I guess

Reply:
Yes, that is true. However, if we don't change the repository, the SPARQL view will not return the correct results. This is also true when we navigate the user to create a similarity index or ChatGPT (similarity and ChatGPT are still in progress).